x-pack/plugins/ml/public/embeddables/types.ts

```diff
@@ -37,9 +43,18 @@ import type {
   MlEmbeddableTypes,
 } from './constants';
 
-export type MlEmbeddableBaseApi = Partial<
-  HasParentApi<PublishesUnifiedSearch> & PublishesViewMode & PublishesUnifiedSearch
->;
+/**
+ * Common API for all ML embeddables
+ */
+export interface MlEmbeddableBaseApi<StateType extends object = object>
+  extends DefaultEmbeddableApi<StateType>,
+    PublishesDataViews,
+    PublishesUnifiedSearch {
+  /**
+   * Result time range based on the parent and panel time range APIs
+   */
+  appliedTimeRange$: PublishingSubject<TimeRange>;
```

Comment:
Why is `appliedTimeRange$` part of this API?  The items in this API are meant to be published externally, whereas the `appliedTimeRange$` is only meant to be used internally to the Embeddable for data fetching when it changes.

Reply:
indeed `appliedTimeRange$` is the result time range used by the embeddable component for data fetching. I reckon it is convenitent to publish externally, so any consumer, e.g. UI action knows what result time range is used, without checking `timeRange$` of the actual embeddable and `parentApi.timeRange$`. Does it make sense?